### ISSUE TITLE
Can't write to HDD anymore

### ISSUE THREAD

**Issue by fogobogo** [midwan/amiberry#351]
### Expected behavior

write to a mounted HDD folder, file.

### Actual behavior

It worked beautifully before so I'm at bit at a loss. 
Amiberry v2.21, tried the -dev branch too.
System is Archlinux ARM on a Raspberry Pi 3b, latest updates, F2FS filesystem. 

Mounted a folder as DF0 etc. Workbench can be installed however, other installer scripts seem to get 0 bytes free upon query and fail accordingly. 
Wing Commander does install but also reports 0 bytes free, others are Syndicate (read / write error or hdd full) 

However, creating / copying a file or folder from within Workbench works fine and Workbench itself reports plenty of space free. Not sure what caused it really.

Creating a Hardfile doesn't work. After poking the code for a bit it seems that the issue is `txtPath->getText().c_str()` returning trashed bytes. I couldn't figure out what caused that.  That `extractPath()` right before that seems sketchy too. Eventually hacked around that.

Either way, same problem with .hdf Hardfiles. 

Basically something is returning 0 bytes free.

**Comment by midwan** [comment]
@fogobogo 
This sounds strange, and I don't believe we've seen it before either.

As @CypherXG mentioned above, please make sure you're not using DF0: as a device name for a hard disk, as that is normally reserved for the internal Floppy drive.

You mentioned "it worked beautifully before", so could you please list what changed from before (the state it worked) until now? Something must have changed, and it wasn't the emulator... :)

**Comment by fogobogo** [comment]
I'm sorry i mixed that up, I indeed meant DH0:

Now if only I knew what actually is different. I'm certainly running out of ideas. I also did check my folder permissions, dabbled with every setting that came to mind, checked the uae configuration that yes, it is indeed rw or days.

reading / writing internally (from Workbench) works without a hitch. Its only installer scripts that seem to fail. Mysterious really.

Creating a Hardfile is broken though. Thats just a plain bug.

**Comment by fogobogo** [comment]
I know. As said its kinda curious. I can drag / drop files from a floppy into a newly created folder in the Workbench and it all shows up on the host folder but any install or setup script fails by claiming there isn't any space left.

**Comment by midwan** [comment]
Is this on a WB 3.1 or WB3.9 installation?
Could you give me specific details about the installer package(s) you tested? I'd like to set up an environment to reproduce this on my end, so I'll need all the steps you took...

**Comment by fogobogo** [comment]
WB3.1 bare bones, no specials, emulated A1200 / AGA /. 2mb Chip / 4mb Fast
I'll attach the uae config renamed to txt because github is picky

[A1200 Deluxe.txt](https://github.com/midwan/amiberry/files/2319302/A1200.Deluxe.txt)
I wouldn't rule out some raspberry pi kernel bug or something similar. If you have any idea where those free space calculations actually happen I could add a few printf's for peeking. Maybe that will get me somewhere.

**Comment by fogobogo** [comment]
Screenshot button would be handy here, tried so far:
Wing Commander (ECS), Syndicate, Amberstar.

Most obvious with Wing Commander, opens a shell reporting 0 bytes free and installs anyway (with success).

**Comment by midwan** [comment]
@fogobogo 
Thanks, let me set up a similar environment here and see what I get.
Meanwhile, you didn't specify which Installer packages you used for testing? I could try a few, but it's best to try the same (as it may work for some and not for others).

**Comment by fogobogo** [comment]
As far as I can tell all those were custom, except WB3.1 itself which came straight from .adf images. 
I just tried the AmiCDROM installer mounted as DH2 with Installer 43.3 in DH0:C/Installer. That worked fine.

**Comment by fogobogo** [comment]
best go with Wing Commander. TOSEC name 'Wing Commander (1992)(Origin)(Disk 1 of 3)[cr Ministry].zip'

**Comment by CypherXG** [comment]
It is the Install script.
I think it cannot handle 3,7GB of free space.
Make an HDF File with 500MB or something else.....install the Game on it and then copy it back to your destination dir where you want the game to play from...

**Comment by midwan** [comment]
What @CypherXG is very likely, as WB3.1 did not have support for very large drives.

**Comment by fogobogo** [comment]
I would tend to agree unless I haven't tried that already. It doesn't explain why I can still copy / write with Workbench (or worked before).

I tried a 100mb HDF file, result was that the Wing Commander installer now reports the right size, however, Syndicate though still ends up the same. So does Amberstar.

Would love to know where UAE / Amiberry does its calculations. I'm convinced _something_ returns the wrong number. Looking at the pacman log the only idea I have is some glibc update.

**Comment by midwan** [comment]
@fogobogo 
Well, it does explain why copy / write with Workbench works actually: WB does not check for free space before the operation starts, if it runs out of space it will tell you so during the process.

Some installers will check for available free space, before attempting to proceed. Others will not (not all installer scripts are the same). Under WB 3.1, the sizes returned are incorrect if we have large partitions, because the OS doesn't support TD64. This is true even on a real Amiga.

**Comment by fogobogo** [comment]
Fair enough, then this would warrant a fix I would assume. I would propose passing a safe number unless I really do run out of space. As it is 0 isn't very practical.

**Comment by fogobogo** [comment]
I finally figured out what prevented me from creating Hardfiles. Turns out `txtPath` and others are deleted before anything is done. I'll attach a diff, it's an easy fix.

[patch.txt](https://github.com/midwan/amiberry/files/2320447/patch.txt)

Short version is that `ExitCreateFilesysHardfile()` is called right before opening a file to actually create a .hdf file.
